performance_regression_test.py

```diff
@@ -790,3 +791,102 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
 
         self.update_hdrhistograms(histogram_name='test_histogram_by_interval',
                                   histogram_data=histogram_data_by_interval)
+
+
+class PerformanceRegressionUpgradeTest(PerformanceRegressionTest, UpgradeTest):  # pylint: disable=too-many-ancestors
+    def __int__(self, *args):
+        super().__init__(*args)
+        self.cluster = None  # pylint: disable=attribute-defined-outside-init
+        self.tester = None  # pylint: disable=attribute-defined-outside-init
+        self.monitoring_set = None  # pylint: disable=attribute-defined-outside-init
+
+    def get_email_data(self):  # pylint: disable=no-self-use
+        """prepare data to generate and send via email
+
+        Have to return the dict which is used to build the
+        html content with email template.
+        Required field:
+        - email_template: path to file with html template
+        - email_subject: subject of email
+        have to be implemented in child class.
+        """
+        return {}
+
+    @latency_calculator_decorator(legend="Upgrade Node")
+    def _upgrade_node(self, node):
+        InfoEvent(message='Upgrade Node %s begin' % node.name).publish()
+        self.upgrade_node(node)
```

Comment:
Please add inside "upgrade_node" info events that will help to breakdown the operation smaller chunks.
Including:
upgrade_system
drain
snapshot
shutdown
packages installations
sysconfig_setup
start node
upgrade sstables

Reply:
added